go-controller/pkg/ovn/routeimport/route_import.go

```diff
@@ -0,0 +1,560 @@
+		t := setTimer(nil, subscribePeriod)
+		for {
+			select {
+			case <-stop:
+				return
+			case <-t.C:
+				if routeEventCh != nil {
+					continue
+				}
+				routeEventCh = subscribeNetlinkRouteEvents(c.netlink, stop, onError)
+				if routeEventCh == nil {
+					t = setTimer(t, subscribePeriod)
+				}
+			case r, open := <-routeEventCh:
+				if !open {
+					c.log.Info("Subscription to netlink route events closed")
+					routeEventCh = nil
+					t = setTimer(t, subscribePeriod)
+					continue
+				}
+				c.log.V(5).Info("Received route event", "event", r)
+				c.syncRouteUpdate(&r)
+			}
+		}
+	}()
+
+	go func() {
+		onError := func(err error) {
+			c.log.Error(err, "Error on netlink link event subscription")
+		}
+		linkEventCh := subscribeNetlinkLinkEvents(c.netlink, stop, onError)
+		t := setTimer(nil, subscribePeriod)
+		for {
+			select {
+			case <-stop:
+				return
+			case <-t.C:
+				if linkEventCh != nil {
+					continue
+				}
+				linkEventCh = subscribeNetlinkLinkEvents(c.netlink, stop, onError)
+				if linkEventCh == nil {
+					t = setTimer(t, subscribePeriod)
+					continue
+				}
+				c.tables = map[int]int{}
+			case l, open := <-linkEventCh:
+				if !open {
+					c.log.Info("Subscription to netlink link events closed")
+					linkEventCh = nil
+					t = setTimer(t, subscribePeriod)
+					continue
+				}
+				c.log.V(5).Info("Received link event", "event", l)
+				c.syncLinkUpdate(&l)
+			}
+		}
+	}()
+}
+
+func (c *controller) syncRouteUpdate(update *netlink.RouteUpdate) {
+	if update.Protocol != unix.RTPROT_BGP {
+		return
+	}
+
+	table := update.Table
+	network := c.getNetworkForTable(table)
+	if network != nil {
+		c.reconcile(network.GetNetworkName())
+	}
+}
+
+func (c *controller) syncLinkUpdate(update *netlink.LinkUpdate) {
+	vrf, isVrf := update.Link.(*netlink.Vrf)
+	if !isVrf {
+		return
+	}
+
+	name := vrf.Name
+	if !strings.HasPrefix(name, types.UDNVRFDevicePrefix) {
+		return
+	}
+	if !strings.HasSuffix(name, types.UDNVRFDeviceSuffix) {
+		return
+	}
+
+	id, err := strconv.Atoi(name[len(types.UDNVRFDevicePrefix) : len(name)-len(types.UDNVRFDeviceSuffix)])
+	if err != nil {
+		c.log.Error(err, "Failed to parse network ID from device name", "name", name)
+		return
+	}
+
+	c.Lock()
+	defer c.Unlock()
+
+	table := int(vrf.Table)
+	network := c.networkIDs[id]
+	info := c.networks[network]
+	current := c.tables[table] == id
+	var old int
+	if info != nil {
+		old = info.table
+	}
+
+	switch update.IfInfomsg.Type {
+	case unix.RTM_DELLINK:
+		if !current {
+			c.log.Info("Ignoring VRF delete for old network", "network", id)
+			return
+		}
+		delete(c.tables, table)
+		table = 0
+	case unix.RTM_NEWLINK:
+		delete(c.tables, old)
+		c.tables[table] = id
+	default:
+		c.log.Info("Unexpected VRF update event type", "type", update.IfInfomsg.Type)
+		return
+	}
+	if info != nil {
+		info.table = table
+	}
+
+	reconcile := info != nil && table != 0 && !current
```

Comment:
nit: s/reconcile/needsReconcile

Reply:
it looks like you are using table 0 to represent an invalid or not found table. Table 0 is a legit table in linux (although it is reserved for unicast addressing). I think it would be better to make a constant called "noTable" or something and use -1.

Reply:
Will do. Just thought that 0 is just as invalid as -1 for us since we are not operating with that table and makes things easier since I don't have to deal with initialization.